From: David Faure <faure@kde.org>
Date: Sun, 26 Jul 2015 14:04:35 +0000
Subject: Add cmake_min_req matching kdelibs req & policies, fix build accordingly.
X-Git-Url: http://quickgit.kde.org/?p=analitza.git&a=commitdiff&h=9c677d2ce90e1c0868f889728a1b765883bbb474
---
Add cmake_min_req matching kdelibs req & policies, fix build accordingly.

CCMAIL: kde-buildsystem@kde.org
---


--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,3 +1,4 @@
+cmake_minimum_required(VERSION 2.8.9)
 project(analitza)
 
 find_package(KDE4 REQUIRED)

--- a/analitza/CMakeLists.txt
+++ b/analitza/CMakeLists.txt
@@ -45,7 +45,7 @@
 )
 
 kde4_add_library(analitza SHARED ${analitza_SRCS})
-target_link_libraries(analitza ${QT_QTCORE_LIBRARY} ${QT_QTXML_LIBRARY} ${KDE4_KDECORE_LIBS})
+target_link_libraries(analitza LINK_PRIVATE ${QT_QTCORE_LIBRARY} ${QT_QTXML_LIBRARY} ${KDE4_KDECORE_LIBS})
 
 set_target_properties(analitza PROPERTIES VERSION ${ANALITZA_LIB_VERSION} SOVERSION ${ANALITZA_LIB_SOVERSION} )
 

--- a/analitzagui/CMakeLists.txt
+++ b/analitzagui/CMakeLists.txt
@@ -22,7 +22,7 @@
 endif(HAVE_OPENGL)
 
 kde4_add_library(analitzagui SHARED ${analitzagui_SRCS})
-target_link_libraries(analitzagui ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${KDE4_KDEUI_LIBS} ${QT_QTSVG_LIBRARY} ${QT_QTOPENGL_LIBRARY} analitza analitzaplot)
+target_link_libraries(analitzagui LINK_PRIVATE ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${KDE4_KDEUI_LIBS} ${QT_QTSVG_LIBRARY} ${QT_QTOPENGL_LIBRARY} analitza analitzaplot)
 
 set_target_properties(analitzagui PROPERTIES VERSION ${ANALITZA_LIB_VERSION} SOVERSION ${ANALITZA_LIB_SOVERSION} )
 

--- a/analitzaplot/CMakeLists.txt
+++ b/analitzaplot/CMakeLists.txt
@@ -66,6 +66,7 @@
 
 kde4_add_library( analitzaplot SHARED ${analitzaplot_SRCS} )
 target_link_libraries ( analitzaplot
+   LINK_PRIVATE
   ${QT_QTCORE_LIBRARY}
   ${QT_QTGUI_LIBRARY}
   ${KDE4_KDECORE_LIBS}
@@ -75,13 +76,14 @@
 
 if(HAVE_OPENGL)
     target_link_libraries ( analitzaplot
+     LINK_PRIVATE
         ${OPENGL_gl_LIBRARY}
         ${OPENGL_glu_LIBRARY}
     )
 endif(HAVE_OPENGL)
 
 if(WIN32)
-    target_link_libraries(analitzaplot ${GLEW_LIBRARIES})
+   target_link_libraries(analitzaplot LINK_PRIVATE ${GLEW_LIBRARIES})
 endif(WIN32)
 
 set_target_properties(analitzaplot PROPERTIES VERSION ${ANALITZA_LIB_VERSION} SOVERSION ${ANALITZA_LIB_SOVERSION} )

--- a/analitzaplot/tests/CMakeLists.txt
+++ b/analitzaplot/tests/CMakeLists.txt
@@ -12,7 +12,7 @@
 kde4_add_unit_test(plotsmodeltest plotsmodeltest.cpp)
 target_link_libraries(plotsmodeltest ${testLibs})
 
-add_definitions("-DSOURCE_DIR=\\\"${CMAKE_CURRENT_SOURCE_DIR}\\\"")
+add_definitions(-DSOURCE_DIR="\\\"${CMAKE_CURRENT_SOURCE_DIR}\\\"")
 kde4_add_unit_test(plotsdictionarymodeltest plotsdictionarymodeltest.cpp)
 target_link_libraries(plotsdictionarymodeltest ${testLibs})
 

